File: bubble_sort/bubble_sort.rb
# Bubble Sort

def bubble_sort(array)
  length = array.length
  return array if length <= 1
  swapped = true
  while swapped do
    swapped = false
    array[0..length-2].each_with_index do |int, idx|
      if array[idx] > array[idx+1]
        array[idx], array[idx+1] = array[idx+1], array[idx]
        swapped = true
      end
    end
  end
  return array
end

arr1 = [4, 1, 7, 9, 3, 2, 6]
arr2 = [8, 7, 6, 5, 4, 3, 2]
arr3 = [-5, -8, 3, 2, -4, -3, 6]
arr4 = [1]
p bubble_sort(arr1)
p bubble_sort(arr2)
p bubble_sort(arr3)

File: bubble_sort/bubble_sort.py
# Bubble Sort

def bubble_sort(list):
  if len(list) <= 1:
    return list
  swapped = True
  while swapped:
    swapped = False
    for i in range(0, len(list)-1):
      if list[i] > list[i+1]:
        list[i], list[i+1] = list[i+1], list[i]
        swapped  = True
  return list



list1 = [4, 1, 7, 9, 3, 2, 6]
list2 = [8, 7, 6, 5, 4, 3, 2]
list3 = [-5, -8, 3, 2, -4, -3, 6]
list4 = [1]
print(bubble_sort(list1))
print(bubble_sort(list2))
print(bubble_sort(list3))
print(bubble_sort(list4))
